packages/account/src/providers/transaction-request/transaction-request.test.ts

```diff
@@ -15,111 +18,143 @@ import { transactionRequestify } from './utils';
+    const address2 = Address.fromRandom();
+
+    const amount1 = 100;
+    const amount2 = 300;
+
+    transactionRequest.addCoinOutput(address1, amount1, ASSET_B);
+    transactionRequest.addCoinOutput(address2, amount2, ASSET_A);
+
+    const result = transactionRequest.getCoinOutputsQuantities();
+
+    expect(result).toEqual([
+      {
+        amount: bn(amount1),
+        assetId: ASSET_B,
+      },
+      {
+        amount: bn(amount2),
+        assetId: ASSET_A,
+      },
+    ]);
   });
 
-  describe('getCoinOutputsQuantities', () => {
-    it('should correctly map all the coin outputs to CoinQuantity', () => {
-      const transactionRequest = new ScriptTransactionRequest();
+  it('should return an empty array if there are no coin outputs', () => {
+    // Mock the getCoinOutputs method
+    const transactionRequest = new ScriptTransactionRequest();
 
-      const address1 = Address.fromRandom();
-      const address2 = Address.fromRandom();
+    const result = transactionRequest.getCoinOutputsQuantities();
 
-      const amount1 = 100;
-      const amount2 = 300;
+    expect(result).toEqual([]);
+  });
+
+  it('should fund with the expected quantities', () => {
+    const transactionRequest = new ScriptTransactionRequest();
 
-      transactionRequest.addCoinOutput(address1, amount1, ASSET_B);
-      transactionRequest.addCoinOutput(address2, amount2, ASSET_A);
+    const amountBase = bn(500);
+    const amountA = bn(700);
+    const amountB = bn(300);
 
-      const result = transactionRequest.getCoinOutputsQuantities();
+    const quantities: CoinQuantity[] = [
+      { assetId: BaseAssetId, amount: amountBase },
+      { assetId: ASSET_A, amount: amountA },
+      { assetId: ASSET_B, amount: amountB },
+    ];
 
-      expect(result).toEqual([
-        {
-          amount: bn(amount1),
-          assetId: ASSET_B,
-        },
-        {
-          amount: bn(amount2),
-          assetId: ASSET_A,
-        },
-      ]);
-    });
+    transactionRequest.fundWithFakeUtxos(quantities);
 
-    it('should return an empty array if there are no coin outputs', () => {
-      // Mock the getCoinOutputs method
-      const transactionRequest = new ScriptTransactionRequest();
+    const inputs = transactionRequest.inputs as CoinTransactionRequestInput[];
 
-      const result = transactionRequest.getCoinOutputsQuantities();
+    const inputA = inputs.find((i) => i.assetId === ASSET_A);
+    const inputB = inputs.find((i) => i.assetId === ASSET_B);
+    const inputBase = inputs.find((i) => i.assetId === BaseAssetId);
 
-      expect(result).toEqual([]);
-    });
+    expect(inputA?.amount).toEqual(bn(700));
+    expect(inputB?.amount).toEqual(bn(300));
+    expect(inputBase?.amount).toEqual(bn(500));
   });
 
-  describe('fundWithFakeUtxos', () => {
-    it('should fund with the expected quantities', () => {
-      const transactionRequest = new ScriptTransactionRequest();
+  it('updates witnesses', () => {
```

Comment:
This is the new test.

Reply:
So the issue I was actually trying to test for seems to be more of a browser performance issue than something wrong with the code, this test matches that of what the user was trying to do, on the level of `ScriptTransactionRequest`. However I am also making changes for #1745 that will impact services above `ScriptTransactionRequest`, but will go into more detail in that PR (when it is ready).